src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs

```diff
@@ -448,11 +442,10 @@ protected virtual CompletionItem GetBetterItem(CompletionItem item, CompletionIt
         private class DisplayNameToItemsMap : IEnumerable<CompletionItem>, IDisposable
         {
             // We might need to handle large amount of items with import completion enabled,
-            // so use a dedicated pool to minimize array allocations.
-            // Set the size of pool to a small number 5 because we don't expect more than a
-            // couple of callers at the same time.
-            private static readonly ObjectPool<Dictionary<string, object>> s_uniqueSourcesPool
-                = new(factory: () => new(), size: 5);
+            // so use a dedicated pool to minimize array allocations. Set the size of pool to a small
+            // number 5 because we don't expect more than a couple of callers at the same time.
+            private static readonly ObjectPool<Dictionary<string, object>> s_uniqueSourcesPool = new(factory: () => new(), size: 3);
+            private static readonly ObjectPool<List<CompletionItem>> s_sortListPool = new(factory: () => new(), size: 3);
```

Comment:
comment says '5', but code says '3' :)

Reply:
Is it really helpful to have our own pools here?  I would expect that just using PooledDictionary would be fine.  There would b an underlying array, but we'd never resize it in practice.

or is thsi because we want to reuse the Dictionary entirely, even if it grew large (something the normal pools don't do).

Reply:
> but we'd never resize it in practice.

I think you are right, but since the `Dictionary<string, object>` might be a commonly used type, and we know the dictionary would get large here, hold on to the these few in our own pool would make sure we don't end up with a shared pool of large dictionaries.